Fixing bug with tabs inside a modal.

charleshimmer opened the pull request:
Materializejs's tabs has a bug where you can't use tabs in a modal unless you re-init the tabs after the modal has been rendered and is visible (See https://github.com/Dogfalo/materialize/issues/1634). This modifies the angular-material modal directive to account for that so you can use tabs in a modal. If a ready callback is provided it will still fire.

charleshimmer committed:
You can't use tabs in a modal unless you re-init the tabs plugin after the modal has been rendered and shown. This fixes that by hooking into the modal ready callback and re-initing the tabs inside the modal.

webbiesdk commented:
Isn't the fix just to pass `function() { $('ul.tabs').tabs(); }` to the relevant modal. 

Doing the fix in every modal, because it might contain tabs, seems overkill.

charleshimmer commented:
Yeah that is one way to fix it, but it seems gross to have to reference DOM and jQuery in every controller you want to support tabs in a modal.  

What if we added an optional attribute to the modal trigger directive to support tabs if the caller needs it.  That way it would save you from writing that jQuery tab initialized code for every modal you needed to have tabs. So something like this:

``` html
 <div modal data-target="#demoModal" tabs="true">
```

It would default to false. Thoughts on that?

webbiesdk commented:
Yeah, referencing DOM and jQuery is not the pretiest thing. 

I like the idea to have an argument `tabs="true"`, could you implement that? Just push it to your master branch, and it will show up in this pull-request.

charleshimmer commented:
Yep, I'll push that up and let you know.

charleshimmer committed:
Changing modal tab support to an toggled by an optional enableTabs attribute on the modal trigger.

Merge remote-tracking branch 'upstream/master'

* upstream/master:
  Changing how the open variable works, now it is set to false when the modal closes. https://github.com/krescruz/angular-materialize/pull/128#issuecomment-191407701
  add support for "weekdays-short"
  Adding some documentation for the new modal functionality (work in progress). #128

Conflicts:
	src/angular-materialize.js

charleshimmer commented:
Ok I have made that change. Can't use the attribute name `tabs` since that is the name of the directive for tabs.

webbiesdk commented:
Looks good. Thanks.

charleshimmer commented:
Great thanks, will you take care of updating the docs or do you want me to do that?

webbiesdk commented:
I would appreciate if you did it.

charleshimmer commented:
Ok I've opened a pull request which documents this new attribute.
